install.sh

```diff
@@ -455,10 +455,18 @@ setup_selinux() {
         rpm_site="rpm-testing.rancher.io"
     fi
 
-    policy_hint="please install:
+    . /etc/os-release
```

Comment:
Can you test to ensure that this file exists before sourcing it? I don't believe it's present on all distros.

Reply:
Perhaps something like

```sh
[ -r /etc/os-release ] && . /etc/os-release
```
alt.

```sh
if [ -r /etc/os-release ] ; then
    . /etc/os-release
fi
```
?

Reply:
> Can you test to ensure that this file exists before sourcing it? I don't believe it's present on all distros.

I can, but can you name a distro that doesn't use it?

/etc/os-release has been a standard for so long now even exceedingly crusty platforms like Debian jessie fully implement it.

the k3s docs say that we support "most modern Linux systems" - and I would argue that any distro without an /etc/os-release doesn't comply with that definition of a 'modern Linux system'.

Reply:
> 
> the k3s docs say that we support "most modern Linux systems" - and I would argue that any distro without an /etc/os-release doesn't comply with that definition of a 'modern Linux system'.

Although I am not the one who requested the check I would say that since `install.sh` is not a run-multiple-times-per-second script where performance trumps safety checks, the extremely small performance (and readability and maintainability) penalty such a check imposes is IMO well worth it to never have to think about it again.

Since a huge selling point for k3s is how easy it is, how fast it is to get up and running I would say it is paramount to have `install.sh` as stable and rock-solid as possible at all times.

Reply:
@joakimr-axis true, shall we make an issue for hardening up the cases where we don't do that already then?

Reply:
I am no maintainer here so it is not for me to decide, but if you ask me: yes!
I am all for fail early and as soon as something crucial happens. In this case here, if the file does not exist we won't get the SUSE variable set and that is all fine so then it should not fail. So fail when it should fail, and do not fail when there is no issue.